torch/_dynamo/source.py

```diff
@@ -36,6 +36,15 @@ def is_constant_source(source):
     return False
 
 
+def is_input_source(source):
+    return source.guard_source() in [
+        GuardSource.LOCAL,
+        GuardSource.GLOBAL,
+        GuardSource.LOCAL_NN_MODULE,
+        GuardSource.GLOBAL_NN_MODULE,
+    ]
```

Comment:
Where is this used? What is an input source?

Reply:
oops, dead code. (FWIW, this used to be the set of sources for which select() worked, but I ended up expanding this set.)